### PR TITLE
Package protobuf

### PR DESCRIPTION
**Pull request by woksin**
Generated protobuf and gRPC stuff now recides in a seperate project

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1115616103797671/1124498256747543)

### CODE REVIEW

File: Protobuf/dolittle/interaction/events.relativity/artifact.proto [hunk 1]
```diff
@@ -9,7 +9,7 @@ package dolittle.events.relativity;
 
 import "system/guid.proto";
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/artifact.proto]: Lets drop the Protobuf at the end - as that was actually inaccurate to begin with

File: Protobuf/dolittle/interaction/events.relativity/claim.proto [hunk 2]
```diff
@@ -7,7 +7,7 @@ syntax = "proto3";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/claim.proto]: Same as earlier - drop .Protobuf from the namespace (I Know, this ripples through - sorry about that)

File: Protobuf/dolittle/interaction/events.relativity/committed_event_stream.proto [hunk 3]
```diff
@@ -11,7 +11,7 @@ import "dolittle/interaction/events.relativity/event_envelope.proto";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/committed_event_stream.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/committed_event_stream_with_context.proto [hunk 4]
```diff
@@ -10,7 +10,7 @@ import "dolittle/interaction/events.relativity/execution_context.proto";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/committed_event_stream_with_context.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/event_envelope.proto [hunk 5]
```diff
@@ -10,7 +10,7 @@ import "dolittle/interaction/events.relativity/property_bag.proto";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/event_envelope.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/event_metadata.proto [hunk 6]
```diff
@@ -12,7 +12,7 @@ import "dolittle/interaction/events.relativity/original_context.proto";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/event_metadata.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/event_source_version.proto [hunk 7]
```diff
@@ -7,7 +7,7 @@ syntax = "proto3";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/event_source_version.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/execution_context.proto [hunk 8]
```diff
@@ -9,7 +9,7 @@ package dolittle.events.relativity;
 import "system/guid.proto";
 import "dolittle/interaction/events.relativity/claim.proto";
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/execution_context.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/open_tunnel.proto [hunk 9]
```diff
@@ -11,7 +11,7 @@ import "dolittle/interaction/events.relativity/tenant_offset.proto";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/open_tunnel.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/original_context.proto [hunk 10]
```diff
@@ -9,7 +9,7 @@ package dolittle.events.relativity;
 import "system/guid.proto";
 import "dolittle/interaction/events.relativity/claim.proto";
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/original_context.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/property_bag.proto [hunk 11]
```diff
@@ -9,7 +9,7 @@ import "system/value.proto";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/property_bag.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/quantum_tunnel.proto [hunk 12]
```diff
@@ -10,7 +10,7 @@ package dolittle.events.relativity;
 import "dolittle/interaction/events.relativity/open_tunnel.proto";
 import "dolittle/interaction/events.relativity/committed_event_stream_with_context.proto";
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Grpc";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Grpc";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/quantum_tunnel.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/tenant_offset.proto [hunk 13]
```diff
@@ -9,7 +9,7 @@ import "system/guid.proto";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/tenant_offset.proto]: Same as before

File: Protobuf/dolittle/interaction/events.relativity/versioned_event_source.proto [hunk 14]
```diff
@@ -10,7 +10,7 @@ import "dolittle/interaction/events.relativity/event_source_version.proto";
 
 package dolittle.events.relativity;
 
-option csharp_namespace = "Dolittle.Runtime.Events.Relativity.Protobuf";
+option csharp_namespace = "Dolittle.Runtime.Grpc.Interaction.Protobuf";
```
Comment (einari) [inline Protobuf/dolittle/interaction/events.relativity/versioned_event_source.proto]: Same as before

File: Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs [hunk 15]
```diff
@@ -88,122 +86,104 @@ public class QuantumTunnelConnection : IDisposable
...
-                    _runCancellationToken.ThrowIfCancellationRequested();
-
-                    try
-                    {
-                        await OpenAndHandleStream();
-                    }
-                    catch (Exception ex)
-                    {
-                        _logger.Error(ex, "Error occurred during establishing quantum tunnel");
+                for (;;) {
```
Comment (einari) [inline Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs]: In C# we don't have our curlies on the same line - but the next line

File: Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs [hunk 16]
```diff
@@ -88,122 +86,104 @@ public class QuantumTunnelConnection : IDisposable
...
-            _logger.Information("Collapsing quantum tunnel");
-            _channel.ShutdownAsync();
+            _logger.Information ("Collapsing quantum tunnel");
+            _channel.ShutdownAsync ();
         }
 
-        void Run()
-        {
-            _logger.Information($"Establishing connection towards event horizon for application ('{_destinationKey.Application}') and bounded context ('{_destinationKey.BoundedContext}') at '{_url}'");
+        void Run () {
```
Comment (einari) [inline Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs]: Curly on next line

File: Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs [hunk 17]
```diff
@@ -88,122 +86,104 @@ public class QuantumTunnelConnection : IDisposable
...
-        {
-            Close();
+        void AssemblyLoadContextUnloading (AssemblyLoadContext context) {
+            Close ();
         }
 
-        void Close()
-        {
-            _runCancellationTokenSource.Cancel();
+        void Close () {
```
Comment (einari) [inline Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs]: Curly should be on next line

File: Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs [hunk 18]
```diff
@@ -88,122 +86,104 @@ public class QuantumTunnelConnection : IDisposable
...
             AssemblyLoadContext.Default.Unloading -= AssemblyLoadContextUnloading;
 
-            Close();
+            Close ();
         }
 
-        void ProcessExit(object sender, EventArgs e)
-        {
-            Close();
+        void ProcessExit (object sender, EventArgs e) {
```
Comment (einari) [inline Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs]: Curly should be on next line

File: Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs [hunk 19]
```diff
@@ -88,122 +86,104 @@ public class QuantumTunnelConnection : IDisposable
...
-            Dispose();
+        ~QuantumTunnelConnection () {
+            Dispose ();
         }
 
         /// <inheritdoc/>
-        public void Dispose()
-        {
+        public void Dispose () {
             AppDomain.CurrentDomain.ProcessExit -= ProcessExit;
```
Comment (einari) [inline Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs]: Curly should be on next line

File: Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs [hunk 20]
```diff
@@ -88,122 +86,104 @@ public class QuantumTunnelConnection : IDisposable
...
+            Console.CancelKeyPress += (s, e) => Close ();
         }
 
         /// <summary>
         /// Destructs the <see cref="QuantumTunnelConnection"/>
         /// </summary>
-        ~QuantumTunnelConnection()
-        {
-            Dispose();
+        ~QuantumTunnelConnection () {
```
Comment (einari) [inline Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs]: Curly should be on next line

File: Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs [hunk 21]
```diff
@@ -65,20 +64,19 @@ public class QuantumTunnelConnection : IDisposable
...
-                FactoryFor<IGeodesics> getGeodesics,
-                FactoryFor<IEventStore> getEventStore,
-                IScopedEventProcessingHub eventProcessingHub,
-                ISerializer serializer,
-                ILogger logger,
-                IExecutionContextManager executionContextManager,
-                ITenants tenants,
-                ITenantOffsetRepository tenantOffsetRepository)
-        {
+        public QuantumTunnelConnection (
```
Comment (einari) [inline Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs]: Curly should be on next line

File: Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs [hunk 22]
```diff
@@ -15,21 +15,20 @@
...
 
-namespace Dolittle.Runtime.Events.Relativity.Grpc
-{
+namespace Dolittle.Runtime.Events.Relativity.Grpc {
     /// <summary>
     /// Represents a concrete connection through a <see cref="IBarrier"/>
     /// </summary>
-    public class QuantumTunnelConnection : IDisposable
-    {
+    public class QuantumTunnelConnection : IDisposable {
```
Comment (einari) [inline Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs]: Curly should be on next line

File: Protobuf/dolittle/interaction/events.relativity/quantum_tunnel.proto (hunk 12 above)
Comment (woksin) [reply Protobuf/dolittle/interaction/events.relativity/quantum_tunnel.proto]: Drop the .Grpc?

File: Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs (hunk 15 above)
Comment (woksin) [reply Source/Events/Relativity/Consumer/QuantumTunnelConnection.cs]: Ahh, I did the "Format Document" command for some of the files, that command apparently did this for some reason...

**Comment by woksin** [comment]
That should cover it, @einari. It also reminded me to configure my c# formater extension to put curly braces on the next line :)

**Comment by einari** [comment]
Tried replying to a comment inside Teams. Not sure if it got here. But keep the Grpc in the namespace and drop Protobuf

**Comment by woksin** [comment]
So I should add back .Grpc?

**Comment by einari** [comment]
Yup. Protobuf is wrong due to it being a message format. Not super enthusiastic about Grpc in the namespace either, but that makes a lot more sense.

**Comment by einari** [comment]
I think we've been misunderstanding each other. The namespace now has `Grpc` twice, it should only have the first `Grpc`

**Comment by woksin** [comment]
But that means that you want it at the time of this commit: https://github.com/dolittle-runtime/Runtime/pull/252/commits/9b3a795fe887626d0db192d29704e56018fa6d44

**Comment by einari** [comment]
Sorry about that..  But yeah..   Complete misunderstanding.  That makes sense.

**Comment by woksin** [comment]
@einari I forced this PR back to the commit I think you wanted, can you see over?

**Pull request merged**